source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,144 @@
+   */
+  uint64_t getRollingValue(absl::string_view cluster_name, absl::string_view stat);
+
+private:
+  /**
+   * Format the given key and absl::string_view value to "key"="value", and adding to the
+   * stringstream.
+   */
+  void addStringToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Format the given key and uint64_t value to "key"=<string of uint64_t>, and adding to the
+   * stringstream.
+   */
+  void addIntToStream(absl::string_view key, uint64_t value, std::stringstream& info);
+
+  /**
+   * Format the given key and value to "key"=value, and adding to the stringstream.
+   */
+  void addInfoToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Generate HystrixCommand event stream.
+   */
+  void addHystrixCommand(std::stringstream& ss, absl::string_view cluster_name,
+                         uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                         uint64_t rolling_window);
+
+  /**
+   * Generate HystrixThreadPool event stream.
+   */
+  void addHystrixThreadPool(std::stringstream& ss, absl::string_view cluster_name,
+                            uint64_t queue_size, uint64_t reporting_hosts, uint64_t rolling_window);
+
+  /**
+   * Building lookup name map for all specific cluster values.
+   */
+  void CreateCounterNameLookupForCluster(const std::string& cluster_name);
+
+  RollingStatsMap rolling_stats_map_;
+  uint64_t current_index_;
+  const uint64_t window_size_;
+  // TODO(trabetti): do we want this to be configurable through the HystrixSink in config file?
+  static const uint64_t DEFAULT_NUM_OF_BUCKETS = 10;
+  std::map<std::string, std::map<std::string, std::string>> counter_name_lookup;
+};
+
+typedef std::unique_ptr<HystrixStatCache> HystrixPtr;
+
+namespace Hystrix {
+
+class HystrixSink : public Stats::Sink, public Logger::Loggable<Logger::Id::hystrix> {
+public:
+  HystrixSink(Server::Instance& server);
+  Http::Code handlerHystrixEventStream(absl::string_view, Http::HeaderMap& response_headers,
+                                       Buffer::Instance&, Server::AdminFilter& admin_filter);
+  void beginFlush() override;
+  void flushCounter(const Stats::Counter& counter, uint64_t delta) override;
+  void flushGauge(const Stats::Gauge&, uint64_t) override{};
+  void endFlush() override;
+  void onHistogramComplete(const Stats::Histogram&, uint64_t) override{};
+
+  // TODO (@trabetti) : support multiple connections
+  /**
+   * register a new connection
+   */
+  void registerConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_register);
+  /**
+   * remove registered connection
+   */
+  void unregisterConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_remove);
+  HystrixStatCache& getStats() { return *stats_; }
+
+private:
+  HystrixPtr stats_;
+  std::vector<Http::StreamDecoderFilterCallbacks*> callbacks_list_{};
+  Server::Instance* server_;
```

Comment:
nit: why not `Server::Instance& server_;` since it never changes what it's pointing to?